server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -251,9 +254,15 @@ export function configureWebSocketServices(
             // Todo: should all the client details come from the claims???
             // we are still trusting the users permissions and type here.
             const messageClient: Partial<IClient> = message.client ? message.client : {};
+            const isSummarizer = messageClient.details?.type === summarizerClientType;
             messageClient.user = claims.user;
             messageClient.scopes = claims.scopes;
 
+            // Do not give SummaryWrite scope to clients that are not summarizers
```

Comment:
i'm not sure this is a good idea. wouldn't it be better for non-summarizer clients to not get a token with this claim? rather than modifying the claims. In the future we plan for features like ondemand summaries, and i'm not sure these will all come from the summarizer client. FYI @pleath and @vladsud

Reply:
I worry about something else - what if we get to the point where main client can summarize (correctly, in presence of local changes). This would be the best path forward (from memory / impact on service, even though expensive in terms of implementation). The fewer there are assumptions in the system that summarizer client is a dedicated client - the better. Someone can just make it happen without asking us :)

Reply:
@anthony-murphy The token is generated by the host even before the client is connected to Fluid. As far as I can tell, summarizer client just inherits the claim from main client? (@vladsud can verify). If that is true, then we have to give every writer "summary" claim because any main client can spawn a summarizer.

@vladsud If the main client becomes the summarizer, then service would have to know it somehow (may be send a flag in connect document call?)

The issue is, we are trying to restrict other clients from sending op once we exceed a certain threshold without summarizing. Today, Push does the same thing by only allowing summary scope to summarizer (also downgrading scopes are okay from a security POV).

FYI @GaryWilber

Reply:
@anthony-murphy Thoughts?

Reply:
We had a discussion about potential alternate approaches to this like accepting only the summary ops but this is not feasible because summarizer also sends other ops like the leader election ops.

We are thinking of merging the current changes in to make it consistent with the logic on push and to make sure that we don't end up accumulating too many ops without summarizing. Please do let me know if there are still concerns with this.